src/components/EditCampaign.js

```diff
@@ -159,25 +162,22 @@ class EditCampaign extends React.Component {
         }
     }
 
-    handleClick = (event) => {
+    handleClick = async (event) => {
         event.preventDefault();
-        if(this.state.updateImage) {
-            this.uploadImageS3(this, this.state.metaFileName);
-            return
-        } else if(this.state.updateDB) {
+        let maxAmountUpdated = await this.updateMaxAmount(this);
+        let imageUpdated = await this.uploadImageS3(this, this.state.metaFileName);
+        if(maxAmountUpdated || imageUpdated){
```

Comment:
If this.updateMaxAmount fails (e.g. it catches an error when transaction fails), maxAmountUpdates will be set to undefined. However, imageUpdated may still be set to true, and we will end up updating the database and showing the user a success message overwriting the error dialog popped up in the "catch" block in "this.updateMaxAmount". We should check maxAmountUpdated before calling this.uploadImageS3, and if maxAmountUpdated is false this function should return w/o trying to make any more updates.